python/ray/_private/log.py

```diff
@@ -1,6 +1,9 @@
 import logging
 import threading
 from typing import Union
+import time
+
+LOG_KEY_TIMESTAMP_NS = "timestamp_ns"
```

Comment:
@jjyao has a slight concern about the collision of `timestamp_ns`, and suggested to use something like `_ray_timestamp_ns` instead.

@GeneDer mentioned that this is a backward incompatible change. Would you mind providing more details?

Reply:
In product, we use `timestamp_ns` field to determine the timestamp at ingestion time and show on the UI. The existing version of Ray is already producing this field and will be used in the log viewer. Yes, we can accept a new field `_ray_timestamp_ns`, but we will always need to support `timestamp_ns` to be compatible with older versions of Ray regardless, and this adds more complexity on the product logic to needing to first check the existence of  `_ray_timestamp_ns` and then check the existence of `timestamp_ns` before populating the timestamp.  

Also, I don't think we should worry about the collision at all since with or without the changes in this PR, `timestamp_ns` is already populating the same way using log record factory. The only difference is before this change, the `timestamp_ns` field is only added when Ray Core's `LoggingConfig` is used, but after this change the `timestamp_ns` field will be added when Ray is imported.